installer/lib/daisy/generator.ex

```diff
@@ -0,0 +1,205 @@
+defmodule Daisy.New.Generator do
+  @moduledoc false
+  import Mix.Generator
+  alias Daisy.New.Project
+
+  @callback prepare_project(Project.t()) :: Project.t()
+  @callback generate(Project.t()) :: Project.t()
+
+  defmacro __using__(_env) do
+    quote do
+      @behaviour unquote(__MODULE__)
+      import Mix.Generator
+      import unquote(__MODULE__)
+      Module.register_attribute(__MODULE__, :templates, accumulate: true)
+      @before_compile unquote(__MODULE__)
+    end
+  end
+
+  defmacro __before_compile__(env) do
+    root = Path.expand("../../templates", __DIR__)
+
+    templates_ast =
+      for {name, mappings} <- Module.get_attribute(env.module, :templates) do
+        for {format, _proj_location, files} <- mappings,
+            format != :keep,
+            {source, _target} <- files,
+            source = to_string(source) do
+          path = Path.join(root, source)
+
+          if format in [:config, :prod_config, :eex] do
+            compiled = EEx.compile_file(path)
+
+            quote do
+              @external_resource unquote(path)
+              @file unquote(path)
+              def render(unquote(name), unquote(source), var!(assigns))
+                  when is_list(var!(assigns)) do
+                unquote(compiled)
+              end
+            end
+          else
+            quote do
+              @external_resource unquote(path)
+              def render(unquote(name), unquote(source), _assigns), do: unquote(File.read!(path))
+            end
+          end
+        end
+      end
+
+    quote do
+      unquote(templates_ast)
+      def template_files(name), do: Keyword.fetch!(@templates, name)
+    end
+  end
+
+  defmacro template(name, mappings) do
+    quote do
+      @templates {unquote(name), unquote(mappings)}
+    end
+  end
+
+  def copy_from(%Project{} = project, mod, name) when is_atom(name) do
+    mapping = mod.template_files(name)
+
+    for {format, project_location, files} <- mapping,
+        {source, target_path} <- files,
+        source = to_string(source) do
+      target = Project.join_path(project, project_location, target_path)
+
+      case format do
+        :keep ->
+          File.mkdir_p!(target)
+
+        :text ->
+          create_file(target, mod.render(name, source, project.binding))
+
+        :config ->
```

Comment:
I think this is not being used right?

Reply:
nope, being used but good to have.

Reply:
[remove unecessary stuff](https://github.com/phcurado/daisy_ui_components/pull/77/commits/4cd5dd9658ff7291b96c19cd1891b74a8ae681df)